### PR TITLE
CIV-9613 Trial Arrangements - CTA “Finalise your Trial or Hearing Arrangements” - Latest Updates Screen

### PR DESCRIPTION
**Pull request by maciejmodzelewski**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9613

### Change description ###
A new tail created with information on the due date of finalising the trial or hearing arrangements and a button that redirects to the “start now” screen to start the process of finalising my trial arrangements (IMPORTANT - the button will not work until CIV-9124 is merged into master)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/claim.ts [hunk 1]
```diff
@@ -546,6 +546,13 @@ export class Claim {
     return new Date(hearingDateTime - threeWeeksMilli).toLocaleDateString('en-GB', options);
   }
 
+  get finalisingTrialArrangementsDeadline(): string {
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/claim.ts:549]: basically we're reusing 
the same code of
  get bundleStitchingDeadline(): string {
    const hearingDateTime = new Date(this.caseProgressionHearing.hearingDate).getTime();
    const threeWeeksMilli = 21 * 24 * 60 * 60 * 1000;
    const options: DateTimeFormatOptions = { day: 'numeric', month: 'long', year: 'numeric' };
    return new Date(hearingDateTime - threeWeeksMilli).toLocaleDateString('en-GB', options);
  }
  
can we reuse the same method?

File: src/main/routes/urls.ts [hunk 2]
```diff
@@ -91,6 +91,8 @@ export const CITIZEN_WHY_DO_YOU_DISAGREE_FULL_REJECTION_URL = `${FULL_REJECTION_
 export const RESPONSE_INCOMPLETE_SUBMISSION_URL = `${BASE_CASE_RESPONSE_URL}/incomplete-submission`;
 export const CITIZEN_FR_AMOUNT_YOU_PAID_URL = `${FULL_REJECTION_URL}/how-much-have-you-paid`;
 export const DEFENDANT_SUMMARY_URL = `${DASHBOARD_URL}/:id/defendant`;
+export const DEFENDANT_DOCUMENTS_URL = `${DASHBOARD_URL}/:id/defendant#documents`;
+
```
Comment (LeonardoPalmeiro) [inline src/main/routes/urls.ts:95]: please remove this blank space.

File: src/main/common/models/claim.ts (hunk 1 above)
Comment (kurt-uk) [reply src/main/common/models/claim.ts:549]: can we also then rename it to what it represents rather to what its used for if were putting in multiple places? something more along the lines of `threeWeeksBeforeHearing()`

Comment (LeonardoPalmeiro) [reply src/main/common/models/claim.ts:549]: My suggestion in here is create another ticket(technical) for refactor this part.

Comment (maciejmodzelewski) [reply src/main/common/models/claim.ts:546]: @kurt-uk it's done.

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by miguelMolina3691 — APPROVED** [review]

**Review by hmcts-version1-ignacio — DISMISSED** [review]

File: src/main/common/utils/pageSectionBuilder.ts
Comment (kurt-uk) [inline src/main/common/utils/pageSectionBuilder.ts:1]: there is a test file for this file, that would be good to add this method to

File: src/main/common/models/claim.ts
Comment (kurt-uk) [inline src/main/common/models/claim.ts:1]: There are tests for this file, you have logic in here that youve added for the finalising trial arrangements

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/common/utils/pageSectionBuilder.ts
Comment (maciejmodzelewski) [reply src/main/common/utils/pageSectionBuilder.ts:1]: Tests added

File: src/main/common/models/claim.ts
Comment (maciejmodzelewski) [reply src/main/common/models/claim.ts:1]: Tests added

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by kurt-uk — DISMISSED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

File: src/main/modules/i18n/locales/en.json [hunk 3]
```diff
@@ -1524,6 +1524,16 @@
           "YOUR_TRIAL_PARAGRAPH": "Your trial has been scheduled for {{ hearingDate }} at {{ hearingTimeHourMinute }} at {{ courtName }}.",
           "YOUR_HEARING_PARAGRAPH": "Your hearing has been scheduled for {{ hearingDate }} at {{ hearingTimeHourMinute }} at {{ courtName }}.",
           "VIEW_HEARING_NOTICE_BUTTON": "View hearing notice"
+        },
+        "FINALISE_TRIAL_ARRANGEMENTS": {
+          "TITLE": "Finalise your trial arrangements for your upcoming trial or hearing (optional)",
+          "FINALISE_TRIAL_ARRANGEMENTS_BUTTON": "Finalise trial arrangements",
+          "DUE_BY": "Due by: Midnight, {{ finalisingTrialArrangementsDeadline }}",
```
Comment (hmcts-version1-ignacio) [inline src/main/modules/i18n/locales/en.json:1531]: This line is different in the other translation file.

Comment (maciejmodzelewski) [reply src/main/modules/i18n/locales/en.json:1531]: You are right. Good call. It's fixed now :-)

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by kurt-uk — DISMISSED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by miguelMolina3691 — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by miguelMolina3691 — APPROVED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Pull request merged**